From f06d4ea2c488181698f709e7d8c1367a5648c8b6 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Sat, 2 Oct 2021 11:24:16 +0200
Subject: [PATCH 08/15] Revert "mm: don't allow executable ioremap mappings"

This reverts commit 8491502f787c4a902bd4f223b578ef47d3490264.

Jailhouse needs executable ioremapped RAM to start the hypervisor.

Upstream-Status: Pending

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
(cherry picked from commit b808e2e02ec18c02084aaf5984e40558ac3f41ee)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 mm/vmalloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 96c105bfdda087..dc4913366d2f85 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -317,7 +317,7 @@ int ioremap_page_range(unsigned long addr, unsigned long end,
 {
 	int err;
 
-	err = vmap_range_noflush(addr, end, phys_addr, pgprot_nx(prot),
+	err = vmap_range_noflush(addr, end, phys_addr, prot,
 				 ioremap_max_page_shift);
 	flush_cache_vmap(addr, end);
 	if (!err)
-- 
2.40.1

